src/index.js

```diff
@@ -30,179 +31,182 @@ type Props = {
+      setPickingColorFromDocument(false)
+      setButtonDisabled(false);
+      document.body.style.cursor = cursorInactive
+    }
+  }, [])
 
-  pickColor = () => {
-    const { onPickStart, once } = this.props
-    const { pickingColorFromDocument } = this.state
-    const { cursorActive } = this
+  // handles button click event to start the action
+  const pickColor = () => {
+    const { onPickStart } = props
 
     if (onPickStart) { onPickStart() }
-    document.body.style.cursor = cursorActive
-    if(!pickingColorFromDocument) {
-      document.addEventListener('click', this.targetToCanvas, { once })
-      this.setState({ pickingColorFromDocument: true })
-    }
+    document.body.style.cursor = cursorActive;
+    setPickingColorFromDocument(true);
+    setButtonDisabled(true);
   }
 
-  targetToCanvas = (e: *) => {
-    const { target } = e
-    const { pickRadius } = this.props
+  const targetToCanvas = (e: *) => {
+    const { pickRadius } = props
+    
+    html2canvas(document.body, { logging: false })
+      .then((canvasEl) => {
+        if (pickRadius === undefined || (pickRadius >= 0 && pickRadius < 1)) {
+          extractColor(canvasEl, e)
+        } else {
+          extractColors(canvasEl, e)
+        }
+      })
 
-    html2canvas(target, { logging: false })
-    .then((canvas) => {
-      if (pickRadius) {
-        this.extractColors(canvas, e)
-      } else {
-        this.extractColor(canvas, e)
-      }
-    })
+    if (props.once === true || props.once === undefined) {
+      setPickingColorFromDocument(false)
+      setButtonDisabled(false);
+      document.body.style.cursor = cursorInactive
+    }
   }
 
-  extractColor = (canvas: *, e: *) => {
-    const { offsetX, offsetY } = e
+  const extractColor = (canvas: *, e: *) => {
+    const { pageX, pageY } = e
+    const pixelColor = getCanvasPixelColor(canvas, pageX, pageY)
+    const { r, g, b } = pixelColor;
 
-    const colors = getCanvasPixelColor(canvas, offsetX, offsetY)
-    this.setColors(colors)
+    updateColors({ r: r, g: g, b: b });
   }
 
-  extractColors = (canvas: *, e: *) => {
-    const { offsetX, offsetY } = e
-    const { unit, amount } = this.props.pickRadius
-
-    let maxRadius, minRadius
-    if (unit === 'radius') {
-      maxRadius = amount
-      minRadius = -(amount) - 1
-    } else if (unit === 'pixel') {
-      if (amount % 2 !== 0) {
-        maxRadius = ((amount - 1) / 2)
-        minRadius = -((amount - 1) / 2) - 1
-      } else {
-        throw new Error('[EyeDrop] The unit \'pixel\' may only have an odd amount.')
-      }
-    } else {
-      throw new Error('[EyeDrop] Please define a proper unit type.')
-    }
+  const extractColors = (canvas: *, e: *) => {
+    const { pageX, pageY } = e
+    const { pickRadius } = props
 
-    const colors = []
-    let radialOffsetX, radialOffsetY
+    const sx = pageX - pickRadius
+    const sy = pageY - pickRadius
+    const sw = pickRadius * 2
+    const sh = pickRadius * 2
```

Comment:
Please give these constants verbose names.
No idea what `sx` or `sh` stands for. :sweat_smile:

Reply:
The `sx`, `sy` names are from the canvas' `getImageData` api, that's what they use:

https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/getImageData

The rectangle starts with position `(sx, sy)` on the axis, and `sw` and `sh`  are the width and height of the rectangle from that starting point.

Reply:
![getImageData](https://user-images.githubusercontent.com/23502533/72914392-e952b600-3d3e-11ea-9f76-ef44e4131a80.png)

Reply:
What about:
xCoord, yCoord, pickWidth, pickHeight
? :smile:

Reply:
w3schools is not the official site for Javascript

Reply:
It may not be, yet it's accurate enough to use in order to grasp verbose naming in terms of Parameter values. 😉

Reply:
The pickWidth and pickHeight name seems right, the x and y need to be something like startingX and startingY to indicate that they are the starting points.

Reply:
Hmm... Yes! That's a good point.
startingX, startingY, pickWidth, pickHeight
it is, then! :smile: